experiment/builder_runner.py

```diff
@@ -358,47 +397,23 @@ def build_and_run(self, generated_project: str, target_path: str,
     coverage_name = f'{uid}.coverage'
     coverage_path = f'gs://{self.experiment_bucket}/{coverage_name}'
 
-    for attempt_id in range(1, CLOUD_EXP_MAX_ATTEMPT + 1):
-      try:
-        sp.run([
-            f'./{oss_fuzz_checkout.VENV_DIR}/bin/python3',
-            'infra/build/functions/target_experiment.py',
-            f'--project={generated_project}',
-            f'--target={self.benchmark.target_name}',
-            f'--upload_build_log={build_log_path}',
-            f'--upload_output_log={run_log_path}',
-            f'--upload_corpus={corpus_path}',
-            f'--upload_coverage={coverage_path}',
-            f'--experiment_name={self.experiment_name}', '--'
-        ] + self._libfuzzer_args(),
-               capture_output=True,
-               check=True,
-               cwd=oss_fuzz_checkout.OSS_FUZZ_DIR)
-        break
-      except sp.CalledProcessError as e:
-        # Replace \n for single log entry on cloud.
-        stdout = e.stdout.decode('utf-8').replace('\n', '\t')
-        stderr = e.stderr.decode('utf-8').replace('\n', '\t')
-
-        captured_error = next(
-            (err for err in self._RETRYABLE_ERRORS if err in stdout + stderr),
-            '')
-        if captured_error and attempt_id < CLOUD_EXP_MAX_ATTEMPT:
-          delay = 5 * 2**attempt_id
-          if captured_error == 'RESOURCE_EXHAUSTED':
-            # Add random jitter in case of exceeding request per minute quota.
-            delay += random.randint(50, 90)
-
-          logging.warning(
-              'Failed to evaluate %s on cloud, attempt %d:\n%s\n%s\n'
-              'Retry in %ds...', os.path.realpath(target_path), attempt_id,
-              stdout, stderr, delay)
-          time.sleep(delay)
-        else:
-          logging.error('Failed to evaluate %s on cloud, attempt %d:\n%s\n%s',
-                        os.path.realpath(target_path), attempt_id, stdout,
-                        stderr)
-          return build_result, None
+    try:
+      self._run_with_retry_control(
+          os.path.realpath(target_path),
+          [
+              f'./{oss_fuzz_checkout.VENV_DIR}/bin/python3',
+              'infra/build/functions/target_experiment.py',
+              f'--project={generated_project}',
+              f'--target={self.benchmark.target_name}',
+              f'--upload_build_log={build_log_path}',
+              f'--upload_output_log={run_log_path}',
+              f'--upload_corpus={corpus_path}',
+              f'--upload_coverage={coverage_path}',
+              f'--experiment_name={self.experiment_name}', '--'
+          ] + self._libfuzzer_args(),
+          cwd=oss_fuzz_checkout.OSS_FUZZ_DIR)
+    except sp.CalledProcessError:
+      return build_result, None
```

Comment:
nit: You don't have to `raise e` in the function and capture it here, if you don't use `e`.
A simple `return True/False` could do the same trick.

Reply:
Done